### PR TITLE
MINOR: improve test coverage for dynamic LogConfig(s)

### PR DESCRIPTION
**Pull request by lbradstreet**
Adding a dynamically updatable log config is currently error prone, as it is easy to set them up as a val not a def and this would result in a dynamically updated broker default not applying to a LogConfig after broker restart.

This PR adds a guard against introducing these issues by ensuring that all log configs are exhaustively checked via a test.

For example, if the following line was a val and not a def, there would be a problem with dynamically updating broker defaults for the config.

https://github.com/apache/kafka/blob/4bde9bb3ccaf5571be76cb96ea051dadaeeaf5c7/core/src/main/scala/kafka/server/KafkaConfig.scala#L1216

### CODE REVIEW

**Comment by lbradstreet** [comment]
Scala 2.11 compile issues are present.

File: core/src/test/scala/unit/kafka/server/KafkaConfigTest.scala
Comment (dhruvilshah3) [inline core/src/test/scala/unit/kafka/server/KafkaConfigTest.scala]: This is nice! Would it be better to consider the configs defined in `LogConfig.TopicConfigSynonyms` only? Maybe something like this:

```
    LogConfig.TopicConfigSynonyms.foreach { case (logConfig, kafkaConfig) =>
        logConfig match {
          case LogConfig.CleanupPolicyProp =>
            assertDynamic(kafkaConfig, Defaults.Compact, () => config.logCleanupPolicy)
          ...
        }
    }
```

**Review by dhruvilshah3 — APPROVED** [review]
Thanks for the PR, LGTM. Left one minor comment.

**Comment by lbradstreet** [comment]
@dhruvilshah3 thanks, your suggestion was great and picked up a couple of issues in the checks. PR is ready to merge when ready.

**Comment by lbradstreet** [comment]
This needs a rebuild by a committer.

**Comment by ijuma** [comment]
ok to test

**Comment by mumrah** [comment]
retest this please

**Comment by lbradstreet** [comment]
One flake in JDK 11 and scala 2.13: `kafka.api.ConsumerBounceTest.testRollingBrokerRestartsWithSmallerMaxGroupSizeConfigDisruptsBigGroup`.

No failures in scala 2.12.

Looks like we're good to merge @ijuma or @mumrah

**Review by ijuma — APPROVED** [review]
LGTM, thanks.

**Pull request merged**
